packages/vite-plugin-angular/src/lib/authoring/ng.ts

```diff
@@ -368,7 +373,8 @@ ${gettersSetters
 function processMetadata(
   metadataObject: ObjectLiteralExpression,
   targetMetadataArguments: ObjectLiteralExpression,
-  targetClass: ClassDeclaration
+  targetClass: ClassDeclaration,
+  targetSourceFile: any
```

Comment:
nit: the type is `SourceFile` imported from `ts-morph`

Reply:
Thanks fixed up the type - as for `export const routeMeta` wasn't there a plan to utilise exports for outputs?

But it does feel a bit strange to me to have two different ways to handle metadata, e.g:

```ts
export const routeMeta = {
  title: 'My page'
}

defineMetadata({
  selector: 'my-component'
})
```

Anyway happy to follow whatever you or Brandon think here, feel free to give me the final word and I'll implement the change

Reply:
re:output No, we're using `new EventEmitter` directly.

I'm ok either way. cc @brandonroberts

Reply:
We do need to find a good pattern for defining code that's inside an ng file that's not component class code.

I was thinking about using `export` but because we can statically find inputs/outputs I don't think its necessary. 

I think for consistency with Analog routes in non-ng files, we should go with the `export const routeMeta` that gets put outside the transformed component class.

Reply:
What about rather than making this change just for `routeMeta` we use `export` as a way to export anything outside of the component class. I've had people complain that this .ng format wouldn't allow doing things like exporting and interface.

So perhaps rather than specifically grabbing the `routeMeta` I make a change to do this:

```html
<script lang="ts">
  export const routeMeta: RouteMeta = {
    title: 'My page'
  }

  export interface MyInterface {
    title: string
  }
</script>

etc
```

```ts
@Component({
  etc
})
export class VirtualAnalogComopnent { 

}

export const routeMeta: RouteMeta = {
  title: 'My page'
}

export interface MyInterface {
  title: string
}
```

Reply:
Yes, I think that's reasonable.

Reply:
I'm also good with that.

Reply:
Pushed an implementation for this - just need to give it a proper look over and will also add an example to the ng-app later today. Let me know if there's anything you want changed.

Reply:
This should be good to go now